src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -322,6 +316,58 @@ struct ScalarOp<ONNXTanOp> {
   using IOp = NotSuportedScalarOp;
 };
 
+//===----------------------------------------------------------------------===//
+// Scalar unary ops for lowering ONNXIsInfOp
+//===----------------------------------------------------------------------===//
+template <>
+struct ScalarOp<ONNXIsInfOp> {
+  using FOp = CustomScalarOp;
+  using IOp = NotSuportedScalarOp;
+};
+
+template <>
+double analyzeSimdFor<ONNXIsInfOp>(Type t, int64_t &von, int64_t &son) {
+  return simdAnalysis({GenericOps::LogicalGop}, {1}, t, von, son);
```

Comment:
I am afraid that we need to have `if/then/else` here to count the number of ops exactly for each case. For example, `detectInf` has two `eq` and one `or`, while `detectPos` and `detectNeg` have one `eq`.

Reply:
Okay let me try and figure it out :)

Reply:
Thanks @AlexandreEichenberger

Reply:
@AlexandreEichenberger So I wrote skeleton code for SIMD as shown below:

```
// pos or neg: 1 compare (1 eq)
// pos and neg: 2 compares (2 eq's) and 1 logical (1 or)

template <>
double analyzeSimdFor<ONNXIsInfOp>(Type t, int64_t &von, int64_t &son) {
  if (pos || neg) {
    return simdAnalysis({GenericOps::CompareGop}, {1}, t, von, son);
  } else {
    return simdAnalysis(
        {GenericOps::CompareGop, GenericOps::LogicalGop}, {2, 1}, t, von, son);
  }
}
```

However, I ran into a slight issue.... the way I distinguish between the positive and negative cases are based on the attribute values which we would retrieve from the `Operation *op`. @tungld suggested we should pass `Operation *op` as a parameter in `analyzeSimdFor` so that we can ensure we have SIMD for the IsInf operation. What are your thoughts? If we add this parameter for `analyzeSimdFor` we would have to change it for every single operation that utilizes the template that was created in `Elementwise.cpp`. If this is a bigger issue, can we merge this PR and I can further work on the enhancement of SIMD for IsInf?

Reply:
@hamptonm1 please add the generic op as parameter. will undoubtedly be useful for other ops as well. that would be a worthy upgrade.

Reply:
@AlexandreEichenberger Okay I added the op as a parameter... let me know if this looks good in `ElementWise.cpp`